feature/lldp/otg_tests/core_lldp_tlv_population_test/core_lldp_tlv_population_test.go

```diff
@@ -0,0 +1,321 @@
+	lldp.SetName(lldpSrc.otgName)
+	lldp.Connection().SetPortName(portName)
+	lldp.ChassisId().MacAddressSubtype().SetValue(lldpSrc.macAddress)
+
+	// Push config and start protocol.
+	otg.PushConfig(t, config)
+	otg.StartProtocols(t)
+
+	return config
+}
+
+// verifyNodeConfig verifies the config by comparing against the telemetry state object.
+func verifyNodeConfig(t *testing.T, node gnmi.DeviceOrOpts, port *ondatra.Port, conf *oc.Lldp, lldpEnabled bool) {
+	statePath := gnmi.OC().Lldp()
+	state := gnmi.Get(t, node, statePath.State())
+	fptest.LogQuery(t, "Node LLDP", statePath.State(), state)
+
+	if lldpEnabled != state.GetEnabled() {
+		t.Errorf("LLDP enabled got: %t, want: %t.", state.GetEnabled(), lldpEnabled)
+	}
+	if state.GetChassisId() != "" {
+		t.Logf("LLDP ChassisId got: %s", state.GetChassisId())
+	} else {
+		t.Errorf("LLDP chassisID is not proper, got %s", state.GetChassisId())
+	}
+	if state.GetChassisIdType() != 0 {
+		t.Logf("LLDP ChassisIdType got: %s", state.GetChassisIdType())
+	} else {
+		t.Errorf("LLDP chassisIdType is not proper, got %s", state.GetChassisIdType())
+	}
+	if state.GetSystemName() != "" {
+		t.Logf("LLDP SystemName got: %s", state.GetSystemName())
+	} else {
+		t.Errorf("LLDP SystemName is not proper, got %s", state.GetSystemName())
+	}
+
+	got := state.GetInterface(port.Name()).GetName()
+	want := conf.GetInterface(port.Name()).GetName()
+
+	if lldpEnabled && got != want {
+		t.Errorf("LLDP interfaces/interface/state/name = %s, want %s", got, want)
+	}
+}
+
+// checkOtgLLDPMetrics verifies OTG side lldp Metrics values based on DUT side lldp is enabled or not
+func checkOtgLLDPMetrics(t *testing.T, otg *otg.OTG, c gosnappi.Config, lldpEnabled bool) {
+	for _, lldp := range c.Lldp().Items() {
+		lastValue, ok := gnmi.Watch(t, otg, gnmi.OTG().LldpInterface(lldp.Name()).Counters().FrameOut().State(), time.Minute, func(v *ygnmi.Value[uint64]) bool {
+			txPackets, _ := v.Val()
+			return v.IsPresent() && txPackets != 0
+		}).Await(t)
+		if !ok {
+			txPackets, _ := lastValue.Val()
+			t.Errorf("LLDP sent packets got: %v, want: > 0.", txPackets)
+		}
+		framesIn, _ := gnmi.Watch(t, otg, gnmi.OTG().LldpInterface(lldp.Name()).Counters().FrameIn().State(), time.Minute, func(v *ygnmi.Value[uint64]) bool {
+			time.Sleep(1 * time.Second)
+			return v.IsPresent()
+		}).Await(t)
+		otgutils.LogLLDPMetrics(t, otg, c)
+		if lldpEnabled {
+			if rxPackets, _ := framesIn.Val(); rxPackets == 0 {
+				t.Errorf("LLDP received packets got: %v, want: > 0.", rxPackets)
+			}
+		} else {
+			if rxPackets, _ := framesIn.Val(); rxPackets != 0 {
+				t.Errorf("LLDP received packets got: %v, want: 0.", rxPackets)
+			}
+		}
+	}
+}
+
+// checkOtgLLDPNeighbors verifies OTG side lldp neighbor states
+func checkOtgLLDPNeighbors(t *testing.T, otg *otg.OTG, c gosnappi.Config, expLldpNeighbors map[string][]lldpNeighbors) {
+	otgutils.LogLLDPNeighborStates(t, otg, c)
+
+	for lldp, lldpNeighbors := range expLldpNeighbors {
```

Comment:
This function feels over-engineered because it's ever only used with just one LLDP neighbor. In the worst case, it has 3 levels of for loops, which means it takes O(n^3) time.

I think if we just ever check for one expected neighbor, then it's just O(n) linear search across the neighbors reported by OTG.

Reply:
Reworked the func to check for the single expected LLDP neighbor